tests/bootstrap.php

```diff
@@ -139,3 +140,7 @@
 // does not allow the sessionid to be set after stdout
 // has been written to.
 session_id('cli');
+
+// Create test database schema.
```

Comment:
Yes it would.

Reply:
That totally sucks. Can't the schema generation be done only when it's actually needed?

Reply:
Plugins tests uses this bootstrap file https://github.com/cakephp/bake/blob/b3b4d1206cd6fbe7b647f67669627f0414b94e42/templates/bake/Plugin/tests/bootstrap.php.twig#L51.

So with currently proposed change even those plugins that have nothing to do with the database would be unnecessarily generating the schema.

Reply:
> That totally sucks. Can't the schema generation be done only when it's actually needed?

Yes. I'm thinking that most plugins will use migrations based schema, and that interface would not reset schema on each run.

We could 'skip' dropping tables that already exists. This would mean that core developers would need to manually wipe their test db when existing fixtures are changed.

Do you want to see how much of an impact clearing and rebuilding the test schema for cakephp's test takes before making a firm decision?

Reply:
> If a plugin is pulling in our tests/bootstrap.php they should probably not do that as it isn't a 'public' interface.

Well they do, the plugin's tests/bootstap.php generated by bake includes the core's tests/bootstrap.php as seen in the template I linked above. It's a good convenience and you don't have to repeat the boiler plate code in each plugin. 

Could we perhaps have a phpunit config based, on which the schema generator usage would be skipped? Then one can either set that config in phpunit.xml or pass at command line.

Reply:
Not to derail the review, but does that means tests/bootstrap.php *is* considered a public api and cannot change in patch releases? I don't think I've seen migration notes on changes to it -- like when we use DB_URL.

Reply:
It is in the gray area of public/internal it seems. I'd prefer it wasn't public but there will likely be lots of plugins that do depend on it to define constants and paths.